src/index.ts

```diff
@@ -150,8 +152,25 @@ module.exports = class ServerlessPlugin {
 
   private async afterDeploy() {
     const config = getConfig(this.serverless.service);
+    const service = this.serverless.service.getServiceName();
+    const env = this.serverless.getProvider("aws").getStage();
+
     if (config.enabled === false) return;
-    return printOutputs(this.serverless);
+    if (config.monitors && config.monitorsApiKey && config.monitorsAppKey) {
```

Comment:
I think we should add some code to the parameters validation logic that if monitors are being created, that the `monitorsApiKey` and `monitorsAppKey` must also set correctly, and display appropriate error messages to the user. For example, if I define monitors but don't specify a valid monitorsApiKey, I think right now the plugin would just silently skip creating monitors. I think it would be better for the deploy to fail in that circumstance to avoid confusion.

We might also want to validate the API key, I'm not sure if we do that for the regular API key but if so we probably want to do it for the monitors API key as well?

Reply:
This might be something that's relevant to the product side of the project so I'll bring it up with the rest of the team.